client/grpc_query.go

```diff
@@ -27,6 +27,16 @@ func (ctx Context) Invoke(grpcCtx gocontext.Context, method string, args, reply
 	if err != nil {
 		return err
 	}
+
+	// parse height header
+	md, _ := metadata.FromOutgoingContext(grpcCtx)
+	if heights := md.Get(grpctypes.GRPCBlockHeightHeader); len(heights) > 0 {
+		height, err := strconv.ParseInt(heights[0], 10, 64)
```

Comment:
why is this not checking the error?

Reply:
If there's any error we don't consider the request header and query using latest height.

Reply:
This means a height was manually provided. I think we should error in such a case where it was provided but is invalid.

Reply:
Okay, that makes sense. I will update